lib/pbench/test/unit/server/query_apis/test_datasets_mappings.py

```diff
@@ -111,4 +111,7 @@ def test_with_db_error(self, client, server_config):
         with client:
             response = client.get(f"{server_config.rest_uri}/datasets/mappings/summary")
             assert response.status_code == HTTPStatus.INTERNAL_SERVER_ERROR
-            assert response.json["message"] == HTTPStatus.INTERNAL_SERVER_ERROR.phrase
+            assert (
+                response.json["message"]
+                == "Unexpected template error: \"Document template for index 'run' not found in the database\""
```

Comment:
Do we really want to return this kind of internal information on an internal server error?  Seems like we would want to log it, but the caller should just know that an "Unexpected error was encountered for the 'datasets/mappings/summary' API", or something like that.

Reply:
That's always a concern. We've had this discussion several times, and I admit I've wavered between the extremes of hiding all internal context or exposing it.

I suppose most recently my primary influence here has been working on functional tests against a "hidden" containerized server where we can't access the logs. Providing more information through the API response is really important in that sort of environment.

In a production environment, the question turns more to the type of users, and specifically we need to be sure we don't give away "secrets" and we don't overwhelm them with details of no relevance to the API client. But in general I also don't see the need to bluster and hide details of what we messed up internally. Is this over the line? Well, I'm not even sure ...

Reply:
Personally, I'm inclined to err on the side of returning more information.  Obviously we don't want to expose "secrets" or anything which could be used to attack our security, and we certainly don't want to overwhelm the recipient with volume; however, for the RESTful API, most of the time humans won't see the error message contents, and the rest of the time it'll probably be useful (as Dave points out, it can get us around barriers like inaccessible server deployments).

Reply:
I would like to go on record as not in favor of returning ANY information on an internal server error.

That information is not part of the API, and we don't want to inadvertently have our clients or external clients come to rely on that information.  Giving a UUID as a context value that one can use to find the particular issue in audit logs or observability logs is sufficient for all use cases.

Reply:
In other words, let's not let our development needs "leak" in to the operational state visible by the API callers.

Reply:
Those are valid points, @portante, and if I had my "OpenVMS upward compatibility forever" hat on, I might be inclined to agree.  However, we don't have the UUID mechanism available, yet, so we cannot rely on that for observability today, and I'm just not that worried about people coming to rely on the content of `500` messages; whereas, being able to easily see what went wrong (particularly prior to releasing v1.0) is of great value.

So, I'm still in the more-is-better camp.

Reply:
Can't we just add a call to `uuid()`?

Reply:
It might be almost that simple.  There are various flavors of UUIDs (`uuid` is a module, not a function; you probably mean `uuid.uuid4()`, but there are other options).

Somebody just has to sit down, decide on the options, and then use it in all the `500` returns (maybe after encapsulating it in some useful way).  So, perhaps not in this PR....

Also, for that approach to be effective, the Pbench developer needs to be able to easily access the Pbench Server log...where are we on that, e.g., with respect to the the canned server?